src/AddItems.js

```diff
@@ -0,0 +1,81 @@
+import React, { useState } from 'react';
+import { db } from './lib/firebase';
+import { capitalize } from './lib/utility';
+
+function AddItems(props) {
+  const [groceryItem, setGroceryItem] = useState('');
+  const [userToken, setUserToken] = useState('randomTokenOne');
+  const [itemFreq, setItemFreq] = useState(7);
+  const [formValidationVisual, setFormValidationVisual] = useState(false);
+
+  const updateGroceryItem = (event) => {
+    setGroceryItem(capitalize(event.target.value));
+    setFormValidationVisual(true);
+  };
+
+  const submitGroceryItem = (event) => {
+    event.preventDefault();
+    const formData = {
+      itemName: groceryItem,
+      frequency: itemFreq,
+      lastPurchaseDate: null,
+    };
+    db.collection(userToken).add({ formData });
+    setGroceryItem('');
+    setItemFreq(7);
+    setFormValidationVisual(false);
+  };
+
+  const radioBtnHandler = (event) => {
+    setItemFreq(+event.target.value);
+    setFormValidationVisual(true);
+  };
+
+  return (
+    <div>
+      <form id="test-form" onSubmit={submitGroceryItem}>
+        <label htmlFor="item">Item name:</label>
+        <input
+          type="text"
+          id="item"
+          value={groceryItem}
+          required={formValidationVisual}
+          onChange={updateGroceryItem}
+        />
+        <input
+          type="radio"
+          id="soon"
+          value={7}
+          name="frequency"
+          checked={itemFreq === 7}
+          required={formValidationVisual}
+          onChange={radioBtnHandler}
```

Comment:
You _may_ need to pass in:
```suggestion
          onChange={(e) => radioBtnHandler(e)}
```

for all of the `onChange` functions.

Reply:
Hey @laurenmbeatty by writing `onChange={radioBtnHandler}` it automatically passes in the event argument, so there's no need to explicitly add it. It's a nice trick to keep the code a little cleaner 😄